### PR TITLE
Check for the MPFR version in RS-based AK

### PR DESCRIPTION
**Pull request by luis4a0**
Since RS does not work with MPFR versions 3.1.4 and 3.1.5, this pull request enforces the version check using a Boost static assertion.
The bug in MPFR is solved in version 3.2 (which did not yet come to light).
Moreover, some checks are added to the AK code in order to quickly identify this kind of bugs.

### CODE REVIEW

**Comment by luis4a0** [comment]
Thanks Laurent! I forgot to add the faulty MPFR version 3.1.3 to the check. I commited bceff2255ff80d6549a6d1f0e7997fe9cc963531 after the PR was merged. Shall I create a new PR?

**Comment by lrineau** [comment]
Just push to the same branch as the commit, and the pull-request will be updated.

**Comment by luis4a0** [comment]
Done.

**Comment by lrineau** [comment]
You must push to your fork repo, usually aliases to `mine` if you have followed CGAL guidelines. Otherwise the PR is not updated.

**Comment by luis4a0** [comment]
Oups, done.

**Comment by lrineau** [comment]
Good. This time we can see the commit bceff2255ff80d6549a6d1f0e7997fe9cc963531 in the Github page of this PR: https://github.com/CGAL/cgal/pull/1572#commits-pushed-bceff22.

File: Algebraic_kernel_d/include/CGAL/RS/rs3_refiner_1.h [hunk 1]
```diff
@@ -104,13 +116,22 @@ operator()
...
+        typedef ZPtraits::Make_square_free              ZSfpart;
+        typedef CGAL::RS_AK1::Signat_1<ZPolynomial,Gmpfr>
+                                                        Signat;
+#endif
         //std::cout<<"refining ["<<left<<","<<right<<"]"<<std::endl;
         // Construct a Gmpz polynomial from the original Gmpq polynomial.
         Polynomial<Gmpz> zpol=CGAL::RS_AK1::Polynomial_converter_1<
                                         CGAL::Polynomial<Gmpq>,
                                         CGAL::Polynomial<Gmpz> >()(qpol);
+#ifndef CGAL_NO_ASSERTIONS
```
Comment (lrineau) [inline Algebraic_kernel_d/include/CGAL/RS/rs3_refiner_1.h]: This should be `#ifndef CGAL_NO_PRECONDITIONS`.

**Comment by luis4a0** [comment]
Done. Pushed.

**Pull request merged**